src/core/sock/sockinfo_tcp.cpp

```diff
@@ -2863,7 +2979,7 @@ int sockinfo_tcp::accept_helper(struct sockaddr *__addr, socklen_t *__addrlen,
     m_ready_conn_cnt--;
     IF_STATS(m_p_socket_stats->listen_counters.n_conn_backlog--);
 
-    remove_received_syn_socket(ns);
+    safe_mce_sys().worker_threads ? assert(m_syn_received.empty()) : remove_received_syn_socket(ns);
```

Comment:
We should ask about entity_context instead of worker_thread. in terms of future we may have seperate threads and entity contexts.